src/main/java/edu/cornell/kfs/module/cg/document/validation/impl/AwardExtensionRule.java

```diff
@@ -223,7 +227,39 @@ protected boolean checkAccountNotUsedOnOtherAwards(AwardAccount awardAccount) {
 
 		return success;
 	}
-	
+
+    /**
+     * Checks for null on the nullable Billing Frequency Code field. Fixes NullPointerException.
+     */
+    @Override
+    protected boolean checkBillingFrequency() {
+        boolean success = true;
+
+        String newBillingFrequencyCode = newAwardCopy.getBillingFrequencyCode();
+        String oldBillingFrequencyCode = oldAwardCopy.getBillingFrequencyCode();
+
+        if (!org.apache.commons.lang3.StringUtils.equals(newBillingFrequencyCode, oldBillingFrequencyCode)) {
+            if (org.apache.commons.lang3.StringUtils.equals(oldBillingFrequencyCode, CGConstants.MILESTONE_BILLING_SCHEDULE_CODE) &&
+                    SpringContext.getBean(AccountsReceivableModuleBillingService.class).hasActiveMilestones(newAwardCopy.getProposalNumber())) {
+                success = false;
+                putFieldError(CGPropertyConstants.AwardFields.BILLING_FREQUENCY_CODE, CGKeyConstants.AwardConstants.ERROR_CG_ACTIVE_MILESTONES_EXIST, getBillingFrequencyDescription(newAwardCopy));
+            } else if (org.apache.commons.lang3.StringUtils.equals(oldBillingFrequencyCode, CGConstants.PREDETERMINED_BILLING_SCHEDULE_CODE) &&
+                    SpringContext.getBean(AccountsReceivableModuleBillingService.class).hasActiveBills(newAwardCopy.getProposalNumber())) {
+                success = false;
+                putFieldError(CGPropertyConstants.AwardFields.BILLING_FREQUENCY_CODE, CGKeyConstants.AwardConstants.ERROR_CG_ACTIVE_BILLS_EXIST, getBillingFrequencyDescription(newAwardCopy));
+            }
+        }
+
+        return success;
+    }
+
+    private String getBillingFrequencyDescription(Award award) {
+        if (award == null || award.getBillingFrequency() == null) {

```

Comment:
I the billingFrequency object null in this case because the new billing frequency code is empty?

Reply:
The NPE was caused by the billing frequency code on the newAward copy being null. I'm not sure what causes the billing frequency code to be null but as it is a nullable field I assumed that we need to handle the null case.